matrix/requests_auth.go

```diff
@@ -9,16 +9,14 @@ import (
 var ErrInvalidToken = errors.New("missing or invalid access token")
 var ErrGuestToken = errors.New("token belongs to a guest")
 
-func GetUserIdFromToken(ctx rcontext.RequestContext, serverName string, accessToken string, appserviceUserId string, ipAddr string) (string, error) {
+func GetUserIdFromToken(ctx rcontext.RequestContext, serverName string, accessToken string, appserviceUserId string, ipAddr string) (userId string, isGuest bool, err error) {
```

Comment:
MMR's unwritten code style (sorry) is to return types, not variables. Explicit `return x, y, z` is easier to follow.

Reply:
4a8ce42